### ISSUE TITLE
vere crashes after replaying events

### ISSUE THREAD

**Issue by mckpm** [urbit/urbit#5046]
**Bug**

`urbit -n 50 pier` replays 50 events and then crashes as follows:

```
pier: (50): play: done
---------------- playback complete ----------------
pier: replay barrier reached, cramming
serf (50): saving rock
hash-cons arena:
  root: B/144
  atoms (4712):
    refs: KB/135.300
    data: MB/14.558.285
    dict: KB/227.392
  total: MB/14.920.977
  cells (2349460):
    refs: MB/70.491.560
    dict: MB/191.691.192
  total: MB/262.182.752
total: MB/277.103.873

address 0 out of loom!
loom: [0x200000000 : 0x280000000)
Assertion '0' failed in noun/events.c:129
                                         home: bailing out
pier: serf unexpectedly shut down
```

**Context**

My ship's host rebooted, leaving me with a corrupt loom.  I moved aside `chk` to replay my event log.  I first tried doing this with `urbit-king`, which made it to event 7768170/8891171 before crashing on a `nest-fail`.  Then I tried replaying with `urbit`, which got to event 1586667/8891171 before crashing thus:

```
allocate: reclaim: half of 3 entries
allocate: reclaim: half of 1 entries
allocate: reclaim: memo cache: empty
home: bailing out
```

I learned about the `-n` switch (thanks, ~monted-tallex!) and started trying to replay part of my event log before this crash, in order to `meld` and continue.  But after playing 1586166 events, `urbit` crashed like this:

```
pier: (1586166): play: done
---------------- playback complete ----------------
pier: replay barrier reached, cramming
serf (1586166): saving rock
hash-cons arena:
  root: B/144
  atoms (297755):
    refs: MB/6.356.220
    data: MB/41.616.502
    dict: MB/17.284.784
  total: MB/65.257.506
  cells (4673737):
    refs: MB/114.057.740
    dict: MB/501.854.056
  total: MB/615.911.796
total: MB/681.169.446

address 0 out of loom!
loom: [0x200000000 : 0x280000000)
Assertion '0' failed in noun/events.c:129
                                         home: bailing out
pier: serf unexpectedly shut down
```

I experimented with smaller `-n` and got the same behavior down to 50 events.  To my surprise, I was able to reproduce with a comet!

**To Reproduce**
1.  Make a comet: `urbit -c mycomet`
2. Move aside `chk`: `mv mycomet/.urb/chk mycomet_original_chk`
3. Try to replay 50 events:  `urbit -n 50 mycomet`

**System**
- OS: linux64
- Vere 1.5

**Comment by mckpm** [comment]
Doesn't reproduce on darwin.

**Comment by zalberico** [comment]
Thanks @mckpm I think this is partially a known issue where event log replay uses up the available memory and then crashes. The workaround is using -n like you realized in small enough batches to avoid this crash. Each successful batch will be the starting point for the next attempt. This is obviously not desirable, but it's a known issue we're tracking.

That said, 50 events is surprising and it's interesting it's not reproducible on darwin. How many events can you run on darwin? Were you able to replay your ship there?

@baudtack - when you recently hit this, what did you have to reduce `-n ` to in order to get things to work.

@joemfb for context.

#6053

**Comment by joemfb** [comment]
This specific output is a null-pointer dereference on `|meld` (`address 0 out of loom!`). I think that it was found and fixed last year, but I'll need to dig around to be sure.

**Comment by zalberico** [comment]
We’re tracking fixes to replay in the urbit vere repo issue #157.
